File: utils/src/main/java/com/lab/azure/bootloader/utils/arguments/PlanArgument.java
package com.lab.azure.bootloader.utils.arguments;

import com.beust.jcommander.Parameter;
import com.beust.jcommander.converters.FileConverter;

import java.io.File;

/**
 * @author willsun
 */
public class PlanArgument {
    @Parameter
    private String task;
    @Parameter(names = "-c")
    private Integer count;
    @Parameter(names = "-s", converter = FileConverter.class)
    private File settingFile;
    @Parameter(names = "-v", required = false)
    private String version = "v1";

    public String getTask() {
        return this.task;
    }

    public Integer getCount() {
        return this.count;
    }

    public File getSettingFile() {
        return this.settingFile;
    }

    public String getVersion() {
        return this.version;
    }
}

File: src/AzureBootloaderCompiler.Utils/Arguments/PlanArgument.cs
using CommandLine;
using CommandLine.Text;
using System;
using System.Collections.Generic;
using System.Text;

namespace AzureBootloaderCompiler.Utils.Arguments
{
    [Verb("plan", HelpText = "生成编译任务, 记入至Table中")]
    class PlanArgument
    {
        [Option('t', "task", Required = true, HelpText = "任务ID")]
        public string Task { get; set; }

        [Option('c', "count", Required = true, HelpText = "编译总个数")]
        public int Count { get; set; }

        [Option('f', "settingFile", Required = true, HelpText = "配置文件")]
        public string SettingFile { get; set; }

        [Option('v', "version", Required = true, HelpText = "版本")]
        public string Version { get; set; }
    }
}
